### PR TITLE
508 popular entries

### PR DESCRIPTION
**Pull request by UlrichKu**
| Q             | A
| ------------- | ---
| Branch        | 6.3.x
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#508
| License       | MIT

### CODE REVIEW

File: src/CoreBundle/Bridge/Chameleon/Module/Sidebar/SidebarBackendModule.php [hunk 1]
```diff
@@ -160,25 +205,88 @@ private function getMenuItems(): array
...
+        $index = array_search($toggledCategoryId, $activeCategoryIds, true);
+        if (false !== $index){
+            unset($activeCategoryIds[$index]);
+        } else {
+            $activeCategoryIds[] = $toggledCategoryId;
+        }
+
+        $session->set(self::OPEN_CATEGORIES_SESSION_KEY, $activeCategoryIds);
+
+        // TODO could/should save this connected to the user (and not login session)?
```
Comment (pixeljunkie) [inline src/CoreBundle/Bridge/Chameleon/Module/Sidebar/SidebarBackendModule.php]: I would reset this on login and not save it in the user object. The current implementation is o.k. for me.

File: src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js [hunk 2]
```diff
@@ -31,7 +32,25 @@
 
             this.$baseElement.on("keyup", this.handleKeyEvent.bind(this));
 
+            this.$baseElement.find("nav").on("scroll", function(evt) {
+                console.log("Setting "+ $(this).scrollTop());
```
Comment (pixeljunkie) [inline src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js]: console.log needs to be removed

File: src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js [hunk 3]
```diff
@@ -31,7 +32,25 @@
...
+                localStorage.setItem('dummy-pos-test', $(this).scrollTop());
+            });
+
             this.$filterElement.focus();
+
+            if (null !== localStorage.getItem('dummy-pos-test')) {
+                setTimeout(function() {$(".sidebar-nav")[0].scrollTop = localStorage.getItem('dummy-pos-test');}, 1000);
+                //$(".sidebar-nav")[0].scrollTop = localStorage.getItem('dummy-pos-test');
```
Comment (pixeljunkie) [inline src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js]: I don`t get what this dummy-pos-test is used for. Should be explained in the code.

File: src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js [hunk 4]
```diff
@@ -182,6 +196,23 @@
...
                 displayState: displayState
             });
+        },
+        onElementClick: function (event) {
+            this.handleElementClickForPopular($(event.target));
+        },
+        handleElementClickForPopular: function($clickedItem) {
+            const clickedMenuId = $clickedItem.data("entry-id");
+
+            console.log(clickedMenuId);
```
Comment (pixeljunkie) [inline src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js]: remove console.log

File: src/CoreBundle/Bridge/Chameleon/Module/Sidebar/SidebarBackendModule.php (hunk 1 above)
Comment (UlrichKu) [reply src/CoreBundle/Bridge/Chameleon/Module/Sidebar/SidebarBackendModule.php]: Comment removed

File: src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js (hunk 4 above)
Comment (UlrichKu) [reply src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js]: Removed

File: src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js (hunk 3 above)
Comment (UlrichKu) [reply src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js]: Code remove / moved to #522

File: src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js (hunk 2 above)
Comment (UlrichKu) [reply src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js]: Code removed

**Pull request merged**